### ISSUE TITLE
Inline Inserter: Allow arrow down from search field

### ISSUE THREAD

**Issue by ianstewart** [WordPress/gutenberg#28670]
It's currently possible to tab out of the search field below and then begin to navigate recommended blocks with arrow keys …

![image](https://user-images.githubusercontent.com/545779/106649300-fc55b400-6556-11eb-91bd-a9757b35ef35.png)

… but it may be useful and natural to also allow using the down arrow key to leave the search field as well.

**Comment by gwwar** [comment]
For folks 👀  this, let's double-check what best practices are for keyboard navigation.

**Comment by mtias** [comment]
I believe this one was requested to work this way before, but I don't fully remember the context.

**Comment by alexstine** [comment]
I couldn't find anything in the WCAG that says this shouldn't be done other than the recommendation of not changing focus to different fields automatically, but it is something I do not agree would benefit usability for screen reader users. The problem with using down arrow to exit a text input field is that is generally how you can have the whole field read back to you. I think Tab is a perfectly acceptable way to navigate in and out of the field and then the inserter can be navigable via arrows. The other issue is this type of interaction is just not common. I think it would catch users off guard that pressing down arrow could lose focus of the input field.

My vote, let's not move forward with this. Don't get me wrong, I do understand the logic behind it, I am just not convinced other users will.

Thanks.

**Comment by annezazu** [comment]
Noting that this came up as part of the [third round of testing for the FSE Outreach Program:](https://make.wordpress.org/test/2021/03/09/fse-program-testing-call-3-create-a-fun-custom-404-page/#comment-1430)

> Block inserter – using the keyboard to type a search and then trying to use the arrow keys to move to select the block doesn’t work. You have to hit TAB first. Is this intentional a11y affordance or an error? I believe you should be able to use down arrow to skip to the list of search results.

**Comment by alexstine** [comment]
I think this would make functionality worse for assistive technology users, here are the 3 proposals I came up with.
1. If writing flow is enabled, then Down Arrow will work to go in to the list of blocks. If it is disabled, only Tab will move to the list of blocks.
2. We can redesign the block inserter that way the input field is rendered as an autocomplete search bar. I am not sure the best way to communicate this though because generally autocomplete search bars you use your Down Arrow to navigate the results but since inserter has a multi-column layout, this will get tricky to communicate. In my opinion it's more work than it's worth.
3. It is left as is.

The issue here is Down Arrow is not a native action to navigate out of text input fields when using screen readers hints the terms "focus" and "edit" mode. Focus mode allows you to navigate the page with a virtual type cursor while edit mode allows you to stay constrained in a text field and jump around using Tab. Breaking this behavior over one keystroke I don't think is worth it.

Would like to keep the conversation going. Any other ideas or thoughts?

Thanks.

**Comment by Mamaduka** [comment]
Should we close this issue with a "left as it is" resolution?

**Comment by youknowriad** [comment]
Closing as not planned. As explained above, Down Arrow in a text field is how screen reader users read the field back, so Tab stays the way to move into the results.
